drivers/bluetooth/hci/hci_spi_st.c

```diff
@@ -334,13 +334,27 @@ static int bt_spi_bluenrg_setup(const struct bt_hci_setup_params *params)
 }
 #endif /* CONFIG_BT_BLUENRG_ACI */
 
-static struct net_buf *bt_spi_rx_buf_construct(uint8_t *msg)
+static int bt_spi_rx_buf_construct(uint8_t *msg, struct net_buf **bufp, uint16_t size)
 {
 	bool discardable = false;
 	k_timeout_t timeout = K_FOREVER;
 	struct bt_hci_acl_hdr acl_hdr;
-	struct net_buf *buf;
-	int len;
+	static int len;
+	struct net_buf *buf = *bufp;
+	int ret = 0;
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_hci_spi_v1)
+	if (buf) {
+		/* Buffer already allocated, waiting to complete event reception */
+		net_buf_add_mem(buf, msg, MIN(size, len-buf->len));
```

Comment:
Coding style (and consistency with yourself): `len - buf->len`

Reply:
Fixed.